core/src/main/java/org/apache/accumulo/core/fate/accumulo/FateMutatorImpl.java

```diff
@@ -132,12 +140,30 @@ public FateMutator<T> delete() {
     return this;
   }
 
+  /**
+   * Require that the transaction status is one of the given statuses. If no statuses are provided,
+   * require that the status column is absent.
+   *
+   * @param statuses The statuses to check against.
+   */
+  public FateMutator<T> requireStatus(TStatus... statuses) {
+    Condition condition = StatusMappingIterator.createCondition(statuses);
+    mutation.addCondition(condition);
+    return this;
+  }
+
   @Override
   public void mutate() {
-    try (BatchWriter writer = context.createBatchWriter(tableName)) {
-      writer.addMutation(mutation);
-    } catch (Exception e) {
-      throw new IllegalStateException(e);
+    try (ConditionalWriter writer = context.createConditionalWriter(tableName)) {
```

Comment:
For this method if there are not conditions, we could use a batch writer for the mutation instead of adding the condition for them empty column.

```suggestion
    if(mutation.getConditions().isEmpty()) {
       // use batch writer
    } else {
       // use conditional writer
    }
```

Reply:
Fixed in d8a3a05